### PR TITLE
only check legacy castling moves

### PR DESCRIPTION
**Pull request by borg323**
Suggested changes for `IsSameMove()` (with tests).

### CODE REVIEW

File: src/chess/board.cc [hunk 1]
```diff
@@ -743,18 +743,21 @@ bool ChessBoard::IsSameMove(Move move1, Move move2) const {
...
-  // already checked for equality).
-  if (!move1_is_castling || !move2_is_castling) return false;
-  // It's two castlings, checking that they are castlings in the same direction.
-  if (move1.to().col() > move1.from().col()) {
-    return move2.to().col() > move2.from().col();
-  }
-  return move2.to().col() < move2.from().col();
+  // Explicitly check all legacy castling moves
+  if (move1.from() != FILE_E) return false;
+  if (move1.to().col() == FILE_A && move2.to().col() == FILE_C &&
```
Comment (mooskagh) [inline src/chess/board.cc]: Looks good, but this one is probably better for readability:

```cpp
for (const auto& x : {{FILE_A, FILE_C}, {FILE_C, FILE_A}, {FILE_H, FILE_G}, {FILE_G, FILE_H}}) {
  if (....) {
    return true;
  }
}
```

Also if `if` is multiline, curly braces are necessary.

And also, I think it's safer to check for own pieces than for rooks, as rooks() also contain opponent's rooks and capture of a opponent rook is a legal move. (not by jumping from file E to A, but still "more legal").

Comment (borg323) [reply src/chess/board.cc]: Actually, I'm not sure we even need to check move legality here or just the representation. So maybe remove both checks for king and rook?

**Comment by borg323** [comment]
I think the last version is clear enough.

**Comment by mooskagh** [comment]
You have to check at least for king.

If there is e.g. rook e1, then e1a1 and e1c1 are different moves.

**Comment by mooskagh** [comment]
Looks good, thanks for helping with that!

**Pull request merged**